lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -414,16 +414,40 @@ SPIRVType *LLVMToSPIRV::transType(Type *T) {
     if (Name == getSPIRVTypeName(kSPIRVTypeName::ConstantPipeStorage))
       return transType(getPipeStorageType(M));
 
-    auto *Struct = BM->openStructType(T->getStructNumElements(), Name.str());
+    const size_t MaxNumElements = 65535 - 2;
```

Comment:
It would be good to reduce amount of magic numbers used in here:

For `2` I suggest to extend `SPIRVTypeStruct` class definition with a filed describing its fixed word count, like it is done for `SPIRVTypePipeStorage `:

https://github.com/KhronosGroup/SPIRV-LLVM-Translator/blob/2f70dccda368494dcd6bae214d793c5c65b44c5e/lib/SPIRV/libSPIRV/SPIRVType.h#L605

For `65535` I suggest to use `INT16_MAX` from [`<cstdint>`](https://en.cppreference.com/w/cpp/types/integer).

Also, this variable can be defined as `constexpr`

Reply:
I agree, but `INT16_MAX` must be `UINT16_MAX`

Reply:
Ok, done.